docs/technical-documentation/rest-create.md

```diff
@@ -0,0 +1,195 @@
+X-Powered-By: PHP/7.1.26-1+ubuntu16.04.1+deb.sury.org+1
+Location: http://localhost:8000/node/3
+Cache-Control: must-revalidate, no-cache, private
+X-UA-Compatible: IE=edge
+Content-language: en
+X-Content-Type-Options: nosniff
+X-Frame-Options: SAMEORIGIN
+Expires: Sun, 19 Nov 1978 05:00:00 GMT
+Vary: 
+X-Generator: Drupal 8 (https://www.drupal.org)
+Content-Length: 1564
+Content-Type: application/json
+
+{"nid":[{"value":3}],"uuid":[{"value":"3f618cdc-3f2a-4e77-b932-9ff1d461a57a"}],"vid":[{"value":3}],"langcode":[{"value":"en"}],"type":[{"target_id":"islandora_object","target_type":"node_type","target_uuid":"62189bec-3ef3-4196-b847-b17e5ce61fd5"}],"revision_timestamp":[{"value":"2019-03-05T18:07:00+00:00","format":"Y-m-d\\TH:i:sP"}],"revision_uid":[{"target_id":1,"target_type":"user","target_uuid":"46a47057-de2d-4ce2-87ae-dbe4551209b8","url":"\/user\/1"}],"revision_log":[],"status":[{"value":true}],"title":[{"value":"Created a collection with POST"}],"uid":[{"target_id":1,"target_type":"user","target_uuid":"46a47057-de2d-4ce2-87ae-dbe4551209b8","url":"\/user\/1"}],"created":[{"value":"2019-03-05T18:07:00+00:00","format":"Y-m-d\\TH:i:sP"}],"changed":[{"value":"2019-03-05T18:07:00+00:00","format":"Y-m-d\\TH:i:sP"}],"promote":[{"value":true}],"sticky":[{"value":false}],"default_langcode":[{"value":true}],"revision_translation_affected":[{"value":true}],"content_translation_source":[{"value":"und"}],"content_translation_outdated":[{"value":false}],"field_access_terms":[],"field_alternative_title":[],"field_description":[],"field_display_hints":[],"field_edtf_date":[],"field_edtf_date_created":[],"field_edtf_date_issued":[],"field_extent":[{"value":"1 item"}],"field_identifier":[],"field_linked_agent":[],"field_member_of":[],"field_model":[{"target_id":23,"target_type":"taxonomy_term","target_uuid":"6a3b293d-4617-417b-99d2-23d75b57f7c2","url":"\/taxonomy\/term\/23"}],"field_pid":[],"field_resource_type":[],"field_rights":[],"field_subject":[]}
+```
+
+The parts of the above request are:
+
+1. `-i` - return the response headers 
+1. `-X POST` - send a POST request
+1. `-u admin:islandora` - use these basic authentication credentials
+1. `-H"Content-type: application/json"` - send the content-type header
+1. `--data {...}` - send the request body (seen above)
+1. `'http://localhost:8000/node?_format=json'` - the endpoint of the request
+
+## Files and Media
+
+Drupal is supposed to have a way to upload files, but this seems to require the use of an X-CSRF-Token, which can only be retrieved using Cookie authentication and even then does not allow you to upload.
+
+Consequently there is a REST endpoint not listed in the REST UI, because it is not configurable and is part of the Islandora 8 system.
+
+This endpoint is available at `http://localhost:8000/node/{node id}/media/{media type}/{media use}`
+
+It only accepts PUT requests. If the media and file don't exist they are created, if they exist the file is updated with the new body.
```

Comment:
How does this check happen?  Is it by file name or by checksum?

Reply:
It's an entity query: https://github.com/Islandora-CLAW/islandora/blob/8.x-1.x/src/IslandoraUtils.php#L425

The MediaSource service calls that function and replaces the contents of the first result with the payload.  That's one of a handful of assumptions in the object model right now being surfaced from https://github.com/Islandora-CLAW/CLAW/issues/1032

Reply:
I added a new line below this one hopefully with a clear explanation of how this is determined.